module/actor/witcherActor.js

```diff
@@ -25,7 +24,266 @@ export default class WitcherActor extends Actor {
         }
       });
     }
+  }
+
+  updateDerived() {
+    const stats = this.system.stats;
+    const base = Math.floor((stats.body.current + stats.will.current) / 2);
+    const baseMax = Math.floor((stats.body.max + stats.will.max) / 2);
+    const meleeBonus = Math.ceil((stats.body.current - 6) / 2) * 2;
+
+    let intTotalModifiers = this.getAllModifier("int").totalModifiers;
+    let refTotalModifiers = this.getAllModifier("ref").totalModifiers;
+    let dexTotalModifiers = this.getAllModifier("dex").totalModifiers;
+    let bodyTotalModifiers = this.getAllModifier("body").totalModifiers;
+    let spdTotalModifiers = this.getAllModifier("spd").totalModifiers;
+    let empTotalModifiers = this.getAllModifier("emp").totalModifiers;
+    let craTotalModifiers = this.getAllModifier("cra").totalModifiers;
+    let willTotalModifiers = this.getAllModifier("will").totalModifiers;
+    let luckTotalModifiers = this.getAllModifier("luck").totalModifiers;
+    let intDivider = this.getAllModifier("int").totalDivider;
+    let refDivider = this.getAllModifier("ref").totalDivider;
+    let dexDivider = this.getAllModifier("dex").totalDivider;
+    let bodyDivider = this.getAllModifier("body").totalDivider;
+    let spdDivider = this.getAllModifier("spd").totalDivider;
+    let empDivider = this.getAllModifier("emp").totalDivider;
+    let craDivider = this.getAllModifier("cra").totalDivider;
+    let willDivider = this.getAllModifier("will").totalDivider;
+    let luckDivider = this.getAllModifier("luck").totalDivider;
+    this.system.stats.int.modifiers.forEach(item => intTotalModifiers += Number(item.value));
```

Comment:
Instead of having a single line for each stat you could use a combination of `Object.keys()`, `reduce` and `map` to get it dynamically.

```javascript
const totalValuesMap = new Map(
  Object.keys(this.system.stats).map(stat => {
    const totalValue = stat.modifiers.reduce((total, mod) => (total += mod.value), this.getAllModifier(stat).totalModifiers);
    return [stat, totalValue];
  })
);

// -> Map with each stat as id and total value for value: [['int', 13], ['cra', 10], ...]
// Access via totalValuesMap.get('int')
```

I didn't test the code, so maybe you need to slightly adjust it. You can also put it into a different structure than a map.

Reply:
This is quite old code and I dont want to refactor everything at once

Reply:
https://github.com/witchertrpg-foundryvtt/TheWitcherTRPG/issues/61